lib/src/helpers/events/providers.dart

```diff
@@ -0,0 +1,627 @@
+// Copyright (c) 2023, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'dart:js_interop';
+import 'dart:js_interop_unsafe';
+
+import '../../../helpers.dart' show Device;
+import '../../../web.dart';
+import 'streams.dart';
+
+/// Statically accessible `EventStreamProvider`s for all event types.
+class EventStreamProviders {
+  static const EventStreamProvider<ProgressEvent> abortEvent =
```

Comment:
please tell me we have a script for this!

Reply:
nope - copied by hand from dart:html. There was a generator there, but we are not carrying that over.